packages/aws-cdk-lib/aws-lambda/lib/function.ts

```diff
@@ -1283,6 +1293,40 @@ Environment variables can be marked for removal when used in Lambda@Edge by sett
     };
   }
 
+  private configureSnapStart(props: FunctionProps): CfnFunction.SnapStartProperty | undefined {
+    // return/exit if no snapStart included
+    if (!props.snapStart) {
+      return undefined;
+    }
+
+    // SnapStart does not support arm64 architecture, Amazon Elastic File System (Amazon EFS), or ephemeral storage greater than 512 MB.
+    // SnapStart doesn't support provisioned concurrency either, but that's configured at the version level,
+    // so it can't be checked at function set up time
+    // SnapStart supports the Java 11 and Java 17 (java11 and java17) managed runtimes.
+    // See https://docs.aws.amazon.com/lambda/latest/dg/snapstart.html
+    if (props.snapStart != SnapStartConf.ON_PUBLISHED_VERSIONS) {
+      throw new Error('SnapStart currently only support published versions');
+    }
+
+    if (props.runtime != Runtime.JAVA_11 && props.runtime != Runtime.JAVA_17 ) {
```

Comment:
@roger-zhangg looks like we're still waiting on some feedback, but pending that the only other change I would make here is to update our `Runtime` class to have `supportsSnapStart` as a `LambdaRuntimeProp`. Then we can add `supportsSnapStart` as true to any relevant `Runtime` when it gets added in the future:

https://github.com/aws/aws-cdk/blob/5fef403825110205d472bae90b63866c850efed6/packages/aws-cdk-lib/aws-lambda/lib/runtime.ts#L3-L26

We could then change this check to `if (!props.runtime.supportsSnapStart) { //... }`